Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/CvarAdapter.cpp

```diff
@@ -0,0 +1,198 @@
+                    [this, functor](const Dom::Path& path, const Dom::Value& value)
+                    {
+                        (*functor)({ ConsoleTypeHelpers::ValueToString(value.GetBool()) });
+                        m_adapter->OnContentsChanged(path, value);
+                    });
+                builder.EndPropertyEditor();
+                return true;
+            }
+            return false;
+        }
+
+        bool BuildEditorForCvar(AdapterBuilder& builder, ConsoleFunctorBase* functor)
+        {
+            // Check all possible type IDs for our supported types and try to create a property editor
+            return TryBuildNumericEditor<int8_t, int16_t, int32_t, int64_t, uint8_t, uint16_t, uint32_t, uint64_t, float, double>(
+                       builder, functor) ||
+                TryBuildStringEditor(builder, functor) || TryBuildMathVectorEditor<AZ::Vector2, 2, Nodes::Vector2>(builder, functor) ||
+                TryBuildMathVectorEditor<AZ::Vector3, 3, Nodes::Vector3>(builder, functor) ||
+                TryBuildMathVectorEditor<AZ::Vector4, 4, Nodes::Vector4>(builder, functor) ||
+                TryBuildMathVectorEditor<AZ::Quaternion, 4, Nodes::Quaternion>(builder, functor) ||
+                TryBuildMathVectorEditor<AZ::Color, 4, Nodes::Color>(builder, functor) || TryBuildCheckBoxEditor(builder, functor);
+        }
+
+        CvarAdapter* m_adapter;
+    };
+
+    CvarAdapter::CvarAdapter()
+        : m_impl(AZStd::make_unique<Impl>(this))
+    {
+    }
+
+    CvarAdapter::~CvarAdapter()
+    {
+    }
+
+    Dom::Value CvarAdapter::GetContents() const
+    {
+        AdapterBuilder builder;
+        builder.BeginAdapter();
+
+        auto consoleInterface = AZ::Interface<AZ::IConsole>::Get();
+        if (consoleInterface != nullptr)
+        {
+            // Enumerate all functors, try to wire up a proper editor for each CVar
+            consoleInterface->VisitRegisteredFunctors(
+                [&](ConsoleFunctorBase* functor)
+                {
+                    // Skip console functors with no registered type ID - CVars all have a type ID set
+                    if (functor->GetTypeId().IsNull())
+                    {
+                        return;
+                    }
+                    builder.BeginRow();
+                    builder.Label(functor->GetName());
+                    if (!m_impl->BuildEditorForCvar(builder, functor))
+                    {
+                        builder.Label("Unknown CVAR type");
+                    }
+                    builder.EndRow();
+                });
+        }
+
+        builder.EndAdapter();
+        return builder.FinishAndTakeResult();
+    }
+
+    void CvarAdapter::OnContentsChanged(const Dom::Path& path, const Dom::Value& value)
+    {
+        // If one of our values has changed, forward a replace patch to the view
+        NotifyContentsChanged({ Dom::PatchOperation::ReplaceOperation(path, value) });
```

Comment:
question: Is it only possible to replace cvar values in a dom? What about add and removes?

Reply:
This adapter doesn't support adding or removing CVARs as it's purely for editing existing ones, so the only patches we'll actually generate are value replacements.